compiler/sigmatch.nim

```diff
@@ -2258,6 +2263,9 @@ proc paramTypesMatch*(m: var TCandidate, f, a: PType,
       # See tsymchoice_for_expr as an example. 'f.kind == tyUntyped' should match
       # anyway:
       if f.kind in {tyUntyped, tyTyped}: result = arg
+      elif m.calleeSym != nil and m.calleeSym.kind in {skMacro, skTemplate} and
```

Comment:
Likely caused by pre-existing code but the manual knows *nothing* about special casing overload resolution for macros and templates. So this is a bug.

Reply:
This basically copied from https://github.com/nim-lang/Nim/blob/33cd883e27f7ba7564cbeb59d2d38fe02bb507bf/compiler/sigmatch.nim#L2061
What are the semantics of a proc like `len` passed to a routine that isn't a macro or template?

Reply:
@Araq

Reply:
Well `len` is a `tyProc`. I don't understand the question I guess.